test/local/metrics/events.js

```diff
@@ -597,15 +609,16 @@ describe('metrics/events', () => {
   })
 
   it('.emit with content-server account.signed event', () => {
-    const metricsContext = mocks.mockMetricsContext()
+    const flowBeginTime = Date.now() - 1
+    const metricsContext = mocks.mockMetricsContext({
+      gather: sinon.spy(() => ({
+        device_id: 'foo',
+        flow_id: 'bar',
+        flowBeginTime
+      }))
+    })
     const request = mocks.mockRequest({
       metricsContext,
-      payload: {
-        metricsContext: {
-          flowId: 'bar',
-          flowBeginTime: Date.now() - 1
-        }
```

Comment:
This was the oversight that meant the unit tests weren't covering the issue; we were always setting metrics context in the request payload. Moving the data in to the `gather` call forces it down the problematic path of execution.